demo/src/main/java/com/example/demo/programmer/ProgrammerController.java

```diff
@@ -0,0 +1,83 @@
+package com.example.demo.programmer;
+
+import com.example.demo.raport.Raport;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/programmer")
+public class ProgrammerController {
+
+    private final ProgrammerRepository programmerRepository;
+
+    @Autowired
+    public ProgrammerController(ProgrammerRepository programmerRepository) {
+        this.programmerRepository = programmerRepository;
+    }
+
+
+    @GetMapping("/getAll")
+    public ResponseEntity<List<Programmer>> getAll(){
+        List<Programmer> result = programmerRepository.findAll();
+        if(result.isEmpty()) {
+            return new ResponseEntity<>(null, HttpStatus.NO_CONTENT);
+        }
+        return new ResponseEntity<>(result, HttpStatus.OK);
+    }
+    @RequestMapping("/findById")
+    public ResponseEntity< Optional<Programmer>> findById(@RequestParam("id") Long id){
+        if(id == null || id<1) {
```

Comment:
Nie wiem czy to potrzebne - najwyżej findById nic nie zwróci

Reply:
Poprawione